data-prepper-plugins/opensearch/src/integrationTest/java/org/opensearch/dataprepper/plugins/sink/opensearch/OpenSearchSinkIT.java

```diff
@@ -969,6 +973,33 @@ public void testEventOutput() throws IOException, InterruptedException {
         sink.shutdown();
     }
 
+    @ParameterizedTest
+    @MethodSource("getAttributeTestSpecialAndExtremeValues")
+    public void testEventOutputWithSpecialAndExtremeValues(final Object testValue) throws IOException, InterruptedException {
+        final String testIndexAlias = "test-alias";
+        final String testField = "value";
+        final Map<String, Object> data = new HashMap<>();
+        data.put(testField, testValue);
+        final Event testEvent = JacksonEvent.builder()
+                .withData(data)
+                .withEventType("event")
+                .build();
+
+        final List<Record<Event>> testRecords = Collections.singletonList(new Record<>(testEvent));
+
+        final PluginSetting pluginSetting = generatePluginSetting(null, testIndexAlias, null);
+        final OpenSearchSink sink = createObjectUnderTest(pluginSetting, true);
+        sink.output(testRecords);
+
+        final List<Map<String, Object>> retSources = getSearchResponseDocSources(testIndexAlias);
+        final Map<String, Object> expectedContent = new HashMap<>();
+        expectedContent.put(testField, testValue);
+
+        assertThat(retSources.size(), equalTo(1));
+        assertThat(retSources.containsAll(Arrays.asList(expectedContent)), equalTo(true));
```

Comment:
Since you are already checking the size, you could do:

```
assertThat(retSources.get(0), equalTo(expectedContent));
```

This way, if the value is incorrect, Hamcrest will include the actual and expected in the error.